0.17/files/docker-entrypoint.sh

```diff
@@ -47,15 +47,15 @@ if [ "$(id -u)" = '0' ]; then
   SU_EXEC="su-exec factorio"
 fi
 
-if ! find -L $SAVES -iname \*.zip -mindepth 1 -print | grep -q .; then
+if ! find -L "$SAVES" -iname \*.zip -mindepth 1 -print | grep -q .; then
   # Generate a new map if no save ZIPs exist
   ${SU_EXEC} /opt/factorio/bin/x64/factorio \
-    --create $SAVES/_autosave1.zip  \
-    --map-gen-settings $CONFIG/map-gen-settings.json \
-    --map-settings $CONFIG/map-settings.json
+    --create "$SAVES/_autosave1.zip" \
+    --map-gen-settings "$CONFIG/map-gen-settings.json" \
+    --map-settings "$CONFIG/map-settings.json"
 fi
 
-exec ${SU_EXEC} /opt/factorio/bin/x64/factorio \
+exec "${SU_EXEC} /opt/factorio/bin/x64/factorio \
```

Comment:
This is going to fail because exec expects its first parameter to be the name of an executable. It does not pass its parameter to a shell or split its parameter on spaces.

Reply:
(Also, even if that did work, the parameters are no longer quoted now.)

Reply:
The original problem was because SU_EXEC was quoted even though it had a value intended to be split on spaces. Either
* SU_EXEC needs to be not quoted (`exec $SU_EXEC /opt/factorio/bin/x64/factorio ...`). This is fine because SU_EXEC is not controlled by the end user, but set to a fixed string which is built to be split on spaces.
* or SU_EXEC must be initially set to `()` instead of empty string, and instead of being set `SU_EXEC="su-exec factorio"`, must be set `SU_EXEC=(su-exec factorio)` and then used as  `exec "${SU_EXEC[@]}" /opt/factorio/bin/x64/factorio ...`. This is a bash array. If it had to contain arbitrary values where some arguments may be user-provided and may contain spaces, this would be the way to go, but otherwise it's a bit overcomplicated.

Reply:
I have just tested and can confirm that @Macil is correct, it fails here.